File: Bit Manipulation/1018. Binary Prefix Divisible By 5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) 
    {
        long sum = 0;
        vector<bool> ans;
        for(auto &val : nums)
        {
            sum += sum + val;
            ans.push_back(sum % 5 == 0);
            sum %= 5;
        }
        return ans;
    }
};

File: Math/1018. Binary Prefix Divisible By 5.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        s = ""
        ans = []
        for ch in nums:
            s += str(ch)
            isDiv = True
            if int(s, 2) % 5:
                isDiv = not isDiv
            ans.append(isDiv)
        return ans